pandas/core/frame.py

```diff
@@ -3159,7 +3161,13 @@ def eval(self, expr, inplace=False, **kwargs):
         kwargs['level'] = kwargs.pop('level', 0) + 1
         if resolvers is None:
             index_resolvers = self._get_index_resolvers()
-            resolvers = dict(self.iteritems()), index_resolvers
+            # column names with spaces are altered so that they can be referred
```

Comment:
would rather like not to do this here. I would actually move the contents of the ``_get_axis_resolvers`` to pandas.core.computation.common and just call it. Note that the import *must* be a local import (as we use numexpr only conditionally). 

This way all of the logic will be together and obvious to future readers. I know this means a bit more refactoring, but will make things better.

Reply:
I am not sure if you mean the `NDFrame._get_axis_resolvers` function. I rather don't touch the code there, since what I implement is just `.eval()/.query()` specific logic. Though, I have moved the logic (of the column_resolvers only) to `common.py` and made it a local import.